### ISSUE TITLE
[ci] Add shellcheck linter

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector-contrib#17279]
### Describe the issue you're reporting

[shellcheck](https://github.com/koalaman/shellcheck) is a static analysis tool for identifying issues on shell scripts. It could be useful for identifying issues like #17037 and it's generally good for enforcing idiomatic and sensible shell scripts.

Running something like:

```sh
shellcheck -x -e SC2059 -e SC2086 .github/workflows/scripts/*.sh internal/buildscripts/*.sh
```

brings up about ~15 issues to be addressed today.

To ensure we continously lint our scripts with shellcheck, we should add a Github workflow that lints our shell scripts with shellcheck and fixes the issues pointed out by the command above.

**Comment by mohitsaxenaknoldus** [comment]
@mx-psi @jpkrohling PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17292

**Comment by jpkrohling** [comment]
@mohitsaxenaknoldus, are you still interested in working on this one?

**Comment by led0nk** [comment]
resolved by #32052

**Comment by mx-psi** [comment]
Thank you @led0nk!
